gateway/blocks_backend.go

```diff
@@ -145,7 +136,8 @@ func NewBlocksBackend(blockService blockservice.BlockService, opts ...BlocksBack
 
 	r = compiledOptions.r
 	if r == nil {
-		fetcher := fetcherConfig.WithReifier(unixfsnode.Reify)
+		// Setup the UnixFS resolver.
+		fetcher := bsfetcher.NewFetcherConfig(blockService)
```

Comment:
Tests are failing (many 404s instead of 200s). I think you're missing the reifier here:

```suggestion
		fetcher := bsfetcher.NewFetcherConfig(blockService).WithReifier(unixfsnode.Reify)
```

Reply:
I mixed PrototypeChooser and Reifier. Yes.